moveit_planners/ompl/ompl_interface/src/parameterization/model_based_state_space.cpp

```diff
@@ -340,7 +340,7 @@ void ompl_interface::ModelBasedStateSpace::copyJointToOMPLState(ompl::base::Stat
 {
   // Copy one joint (multiple variables possibly)
   memcpy(getValueAddressAtIndex(state, ompl_state_joint_index),
-         robot_state.getVariablePositions() + joint_model->getFirstVariableIndex() * sizeof(double),
+         robot_state.getVariablePositions() + joint_model->getFirstVariableIndex(),
```

Comment:
Good catch, when you use `operator+` on a pointer it advances based on the type of the pointer.  Since getVariablePositions returns a double you want to advance the pointer by amounts of `sizeof(double)` not `sizeof(double) * sizeof(double)`.

Reply:
Ah, now I really understand it, finally. Thanks!